plugins/dell-dock/fu-dell-dock-plugin.c

```diff
@@ -75,9 +75,57 @@ fu_dell_dock_plugin_get_dock_type(FuDevice *device, GError **error)
 }
 
 static gboolean
-fu_dell_dock_plugin_probe_ec_v1_subcomponents(FuPlugin *plugin, FuDevice *ec_device, GError **error)
+fu_dell_dock_plugin_probe_ec_v1_mst(FuPlugin *plugin,
+				    FuDevice *ec_device,
+				    guint8 dock_type,
+				    GError **error)
 {
 	FuContext *ctx = fu_plugin_get_context(plugin);
+	const gchar *instance_id;
+	g_autoptr(FuDellDockMst) mst_device = fu_dell_dock_mst_new(ctx);
+
+	instance_id = fu_dell_dock_get_instance_id(dock_type, dock_component_mst, 0, 0);
+	fu_device_add_instance_id(FU_DEVICE(mst_device), instance_id);
+	fu_device_add_child(FU_DEVICE(ec_device), FU_DEVICE(mst_device));
```

Comment:
Warning message is seen at add_child time if no pre-assigned guid:
```
FuEngine             no GUIDs for device VMM5331 in Dell dock [106be225d4ab81a272bee0ae24a9fab93a54edfe]
```

Reply:
If you do fu_device_probe() before adding the child does that fix things? The lifecycle of the dock components is very hard to understand.

Reply:
adding `guid` manually is the old way it used to mitigate this issue, which is missing in this PR. i.e.: https://github.com/fwupd/fwupd/blob/main/plugins/dell-dock/fu-dell-dock-plugin.c#L104.

in upcoming k2 I put the relationship in device->setup:
https://github.com/fwupd/fwupd/pull/7355/files#diff-11355388afc091f3fc67e4b4be9bf11e3aacd8fcc5c5a812ca767e1e463ce236R66